src/connector/src/parser/unified/util.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::error::{ErrorCode, RwError};
+
+use super::{AccessError, ChangeEvent};
+use crate::parser::{SourceStreamChunkRowWriter, WriteGuard};
+
+pub fn apply_row_operation_on_stream_chunk_writer(
+    row_op: impl ChangeEvent,
+    writer: &mut SourceStreamChunkRowWriter<'_>,
+) -> std::result::Result<WriteGuard, RwError> {
+    match row_op.op()? {
+        super::ChangeEventOperation::Upsert => writer.insert(|column| {
+            let res = row_op.access_field(&column.name, &column.data_type);
```

Comment:
For some change log formats, it seems we have to call `op()` again in `access_field`. Can we avoid this duplicated call?

Reply:
BTW, since we always treat `Update` as `Upsert`, the `before` of a Debezium record is always dropped here. I'm not sure whether this matters as we're not respecting the retractable property of the original change log. cc @st1page @tabVersion

Reply:
https://github.com/risingwavelabs/risingwave/issues/9267#issuecomment-1560447220